src/main/scala/djinni/YamlGenerator.scala

```diff
@@ -192,46 +193,54 @@ class YamlGenerator(spec: Spec) extends Generator(spec) {
 }
 
 object YamlGenerator {
-  def metaFromYaml(td: ExternTypeDecl) = MExtern(
+  def metaFromYaml(td: ExternTypeDecl, spec: Spec) = MExtern(
     td.ident.name.stripPrefix(td.properties("prefix").toString), // Make sure the generator uses this type with its original name for all intents and purposes
     td.params.size,
     defType(td),
     td.body,
     MExtern.Cpp(
-      nested(td, "cpp")("typename").toString,
-      nested(td, "cpp")("header").toString,
-      nested(td, "cpp")("byValue").asInstanceOf[Boolean]),
+      nested(td, spec.cppOutFolder.isDefined, "cpp", "typename", _.toString),
+      nested(td, spec.cppOutFolder.isDefined, "cpp", "header", _.toString),
+      nested(td, spec.cppOutFolder.isDefined, "cpp", "byValue", _.asInstanceOf[Boolean])),
```

Comment:
IntelliJ IDEA is always complaining to me that in Scala boolean parameters should be named:
```suggestion
      nested(td, isRequired: spec.cppOutFolder.isDefined, "cpp", "typename", _.toString),
      nested(td, isRequired: spec.cppOutFolder.isDefined, "cpp", "header", _.toString),
      nested(td, isRequired: spec.cppOutFolder.isDefined, "cpp", "byValue", _.asInstanceOf[Boolean])),
```

I kind of like that rule. Maybe you want to add the names?

Reply:
I do like it too. I tried using IntelliJ IDEA the other day but I didn't find an easy way to set up the project (e.g. for running the tests). I'll see if I can configure my warnings in a similar way or have IntelliJ behave :P

Reply:
That's weird, for me it worked out of the box. Maybe you need to install the scala plugin? 
For building and testing I use the integrated sbt shell in the IDE, but executing a specific test from the `>` icon on the test method should work as well.